test/js/api/domain/post/files-spec.js

```diff
@@ -0,0 +1,239 @@
+
+    var defaultApi = 'defaultApi',
+        debugApi = 'debugApi',
+        otherApi = 'otherApi';
+        
+    var defaultPath = 'defaultPath',
+        debugPath = 'debugPath',
+        otherPath = 'otherPath';
+    
+    var defaultData = 'defaultData',
+        debugData = 'debugData',
+        otherData = 'otherData';
+    
+    var defaultSource = 'defaultSource',
+        debugSource = 'debugSource',
+        otherSource = 'otherSource';
+    
+    var defaultScenario = 'defaultScenario',
+        debugScenario = 'debugScenario',
+        otherScenario = 'otherScenario';    
+    
+    before(function() {
+        var normalizeStub = sinon.stub(path, 'normalize');
+        normalizeStub.withArgs(defaultPath).returns(defaultSource);
+        normalizeStub.withArgs(debugPath).returns(debugSource);
+        normalizeStub.withArgs(otherPath).returns(otherSource);
+        
+        var joinStub = sinon.stub(path, 'join');
+        joinStub.withArgs(defaultData, defaultSource).returns(defaultPath);
+        joinStub.withArgs(debugData, debugSource).returns(debugPath);
+        joinStub.withArgs(otherData, otherSource).returns(otherPath);
+        
+        var gatherFilesStub = sinon.stub();        
+        gatherFilesStub.withArgs(defaultSource).returns(getFiles(['a', 'b', 'c']));
+        gatherFilesStub.withArgs(debugSource).returns(getFiles(['d', 'e', 'f']));
+        gatherFilesStub.withArgs(otherSource).returns(getFiles(['g', 'h', 'i']));        
+        files.__set__('gatherFiles', gatherFilesStub);
+                
+        stubs.push(normalizeStub);        
+        stubs.push(joinStub);            
+        // no need to push the gatherFilesStub, since it doesn't override the actual object
+    });
+    
+    after(function() {
+        for (key in stubs) {
+            stubs[key].restore();
+        }            
+    });    
+        
+    var httpRes = function() {
+        return testUtil.mockHttpRes();
+    };
+    
+    var httpReq = function(options) {
+        options = options || {};
+        return testUtil.mockHttpReq('test_req', options.filename || 'request_0', options.api);
+    }; 
+    
+    var scenarios = {
+        'active': '',
+        'other': {
+            'source': 'proxy/post/other'
+        },
+        'debug': {
+            'source': 'proxy/debug/post',
+            'href': 'http://myapp.com/api/post?debug=true&'
+        }
+    };    
+    
+    var getFiles = function(filenames) {
+        filenames = filenames || [];
+        
+        var files = [];
+        
+        _.each(filenames, function(filename) {
+            files.push({'name':filename, 'file':filename});
+        });
+        
+        return files;
+    }
+    
+    var respond = function(req, res, options) {
+        files.__set__('globalConfig', options);                
+        files(req, res, options);            
+    };    
+            
+    it('should populate config for the default scenario', function() {
```

Comment:
It doesn't populate the config, it gather the files according to the scenario configuration. It still expects the config to be as in "default.json" and returns an object with the list of scenarios for the api, the config of the api (as sent in the body of the request) and an object per scenario with an array of the requests available for that scenario. The config file is left unchanged